t/integration/test_tasks.py

```diff
@@ -328,3 +329,16 @@ def test_asyncresult_get_cancels_subscription(self, manager):
 
         new_channels = [channel for channel in get_active_redis_channels() if channel not in channels_before_test]
         assert new_channels == []
+
+    def test_redis_connection_cleanup_with_send_task(self, manager):
```

Comment:
I suggest we mark the test as `xfail`ed and merge this PR.
There should be a matching issue for this PR so that we'll know this isn't yet fixed.

Reply:
@thedrow I don't think that merging this before fixing this is helpful. There are already multiple issues pointing towards a connection leak:

- https://github.com/celery/celery/issues/6963
- https://github.com/celery/celery/issues/6819

What I would I need is some pointers / help. Besides that I'd be more than happy to investigate and help with a fix.

Reply:
> @thedrow I don't think that merging this before fixing this is helpful. There are already multiple issues pointing towards a connection leak:
> 
>     * [Redis result consumer is not drained periodically (memory leak and possible connection reset) #6963](https://github.com/celery/celery/issues/6963)
> 
>     * [Redis result backend connections leak #6819](https://github.com/celery/celery/issues/6819)
> 
> 
> What I would I need is some pointers / help. Besides that I'd be more than happy to investigate and help with a fix.

that should be the spirit